activestorage/lib/active_storage/attached/macros.rb

```diff
@@ -87,11 +104,24 @@ def #{name}
         end
 
         def #{name}=(attachables)
-          #{name}.attach(attachables)
+          @active_storage_attached_#{name}_change =
+            if attachables.nil? || Array(attachables).none?
+              ActiveStorage::Attached::Changes::DeleteMany.new("#{name}", self)
+            else
+              ActiveStorage::Attached::Changes::CreateMany.new("#{name}", self, attachables)
+            end
+        end
+
+        def #{name}_change
+          @active_storage_attached_#{name}_change
+        end
+
+        def clear_#{name}_change
+          @active_storage_attached_#{name}_change = nil
```

Comment:
It might be nice to replace these ivars with a private/nodoc'ed `attachment_changes` hash in the vein of Active Model's `changes`. Then the change classes could use that and skip the need for `public_send`.

Reply:
Agree. I'll hook that up.